server/src/main/java/org/elasticsearch/action/termvectors/TransportShardMultiTermsVectorAction.java

```diff
@@ -80,9 +90,43 @@ protected boolean resolveIndex(MultiTermVectorsShardRequest request) {
 
     @Override
     protected ShardIterator shards(ProjectState project, InternalRequest request) {
-        ShardIterator shards = clusterService.operationRouting()
+        ShardIterator iterator = clusterService.operationRouting()
             .getShards(project, request.concreteIndex(), request.request().shardId(), request.request().preference());
-        return clusterService.operationRouting().useOnlyPromotableShardsForStateless(shards);
+        if (iterator == null) {
+            return null;
+        }
+        return ShardIterator.allSearchableShards(iterator);
+    }
+
+    @Override
+    protected void asyncShardOperation(
+        MultiTermVectorsShardRequest request,
+        ShardId shardId,
+        ActionListener<MultiTermVectorsShardResponse> listener
+    ) throws IOException {
+        if (DiscoveryNode.isStateless(clusterService.getSettings())) {
```

Comment:
I suppose we can capture this only once instead of reevaluating for every request?